cache.go

```diff
@@ -87,15 +90,37 @@ func (cd *cachedData) flush() {
 		return
 	}
 
-	file, err := os.OpenFile(cd.Path, os.O_RDWR|os.O_TRUNC, 0666)
+	file, err := fileutil.LockFile(cd.Path, os.O_RDWR, 0666)
 	if err != nil {
 		log.Printf("Cannot write to the cache file: %v", err)
```

Comment:
I don't think that blocking is a good idea. We should try to do something like a `syscall.FcntlFlock` with `F_SETLK`, that would return with a `EACCES` or `EAGAIN` in that case. Otherwise, as you said, the user would be left with just a black screen....